### PR TITLE
WIP: Refactor python installs from jenkins_worker to jenkins_common

### PR DESCRIPTION
**Pull request by bmedx**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

File: playbooks/roles/jenkins_common/tasks/main.yml [hunk 1]
```diff
@@ -10,17 +12,6 @@
     - install
     - install:system-requirements
 
-- name: Install specific versions of python
-  apt:
-    name: '{{ item }}'
-    state: present
-    update_cache: yes
-  with_items: '{{ jenkins_common_python_versions }}'
-  tags:
```
Comment (bmedx) [inline playbooks/roles/jenkins_common/tasks/main.yml:19]: These tags are lost in the current set of changes, this is probably bad.

File: playbooks/roles/jenkins_common/tasks/python.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@
 # newer
 - name: add deadsnakes PPA for newer Python versions
   apt_repository:
-    repo: "ppa:fkrull/deadsnakes"
+    repo: "ppa:deadsnakes/ppa"
```
Comment (bmedx) [inline playbooks/roles/jenkins_common/tasks/python.yml:7]: This is supposedly the best PPA to use for deadsnakes these days.

File: playbooks/roles/jenkins_worker/tasks/main.yml [hunk 3]
```diff
@@ -7,9 +7,9 @@
 # All jenkins workers
 - include: packer.yml
 - include: system.yml
-- include: python.yml
+- include: python_requests.yml
```
Comment (bmedx) [inline playbooks/roles/jenkins_worker/tasks/main.yml:10]: I made an assumption here that `jenkins_worker` inherits from `jenkins_common` and will have those tasks run before this, I have no idea if that's actually true.

Comment (estute) [reply playbooks/roles/jenkins_worker/tasks/main.yml:10]: It is not true. `jenkins_common` is the base role inherited by any play that sets up a jenkins master instance, i.e. `jenkins_build`. `jenkins_worker` is unrelated and is used to provision an ec2 box to be able to run as a jenkins worker. This means installing the jdk, docker, and in the case of running tests for the edx-platform, a pre-populated virtualenv.

File: playbooks/roles/jenkins_common/tasks/main.yml (hunk 1 above)
Comment (estute) [reply playbooks/roles/jenkins_common/tasks/main.yml:19]: Can you conditionally `include` on the presence of tags?

File: playbooks/roles/jenkins_common/tasks/python.yml (hunk 2 above)
Comment (estute) [reply playbooks/roles/jenkins_common/tasks/python.yml:7]: You may want to check w/ @feanil just to make sure this doesn't have implications on his team's python upgrade work

File: playbooks/roles/jenkins_worker/tasks/main.yml (hunk 3 above)
Comment (bmedx) [reply playbooks/roles/jenkins_worker/tasks/main.yml:10]: Hmm. So I guess that breaks this PR. Is there a better place to put common Python install code than where I stuck it in `jenkins_common`?

**Pull request closed without merging**